backend/src/main/java/eventlistener/service/mail/MailService.java

```diff
@@ -0,0 +1,58 @@
+package eventlistener.service.mail;
+
+import eventlistener.model.event.EventContentDTO;
+import eventlistener.model.notificationuser.NotificationUser;
+
+import org.springframework.mail.MailSendException;
+
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.stereotype.Service;
+import org.thymeleaf.TemplateEngine;
+import org.thymeleaf.context.Context;
+
+import javax.mail.MessagingException;
+import java.util.List;
+
+@Service
+public class MailService {
+
+    private final TemplateEngine templateEngine;
+    private final JavaMailSender emailSender;
+
+    public MailService(JavaMailSender emailSender, TemplateEngine templateEngine) {
+        this.emailSender = emailSender;
+        this.templateEngine = templateEngine;
+    }
+
+    public void handleMailAction(EventContentDTO eventDetails, List<NotificationUser> users){
+        for (NotificationUser user : users) {
+            String content = templateEngine.process("event", setMailContext(user, eventDetails));
+            String subject = eventDetails.getEnvironmentName() + " " + " Notification";
+            try{
+                sendMail(subject, content, user);
+            }catch (MessagingException e) {
+                throw new MailSendException("Failure sending Mail for Event with ID: "+eventDetails.getId(), e);
```

Comment:
```suggestion
                throw new MailNotSendException("Failure sending Mail for Event with ID: "+eventDetails.getId(), e);
```
or maybe ErrorWhileSendingMailException